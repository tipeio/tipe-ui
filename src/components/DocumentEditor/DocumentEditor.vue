<template>
  <tipe-document-block-list
    :data-tipe-ui="$options.name"
    :blocks="blocks"
    :options="options"
    :disabled="disabled"
  />
</template>

<script>
import vueTypes from 'vue-types'
import interfaces from '@tipe/tipe-interfaces'
import TipeDocumentBlockList from './DocumentBlockList'
import TipeDocumentNameInput from './DocumentNameInput'

export default {
  name: 'TipeDocumentEditor',
  components: {
    TipeDocumentBlockList,
    TipeDocumentNameInput
  },
  props: {
    document: vueTypes.shape(interfaces.document),
    disabled: vueTypes.bool.def(false)
  },
  computed: {
    blocks: props =>
      props.document.blocks.map(block => ({
        ...block,
        disabled: props.disabled
      }))
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeDocumentEditor'] {
  height: 100%;
  width: 100%;
  overflow: hidden;
}

.block-list-container {
  padding-top: 2rem;
  padding-bottom: 10rem;
}
</style>
