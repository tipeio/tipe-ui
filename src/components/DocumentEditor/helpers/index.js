export { default as getBlockValueComponent } from './getBlockValueComponent'
