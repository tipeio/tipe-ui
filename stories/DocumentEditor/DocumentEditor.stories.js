import { storiesOf } from '@storybook/vue'
import TipeDocumentBlockList from '@/components/DocumentEditor/DocumentBlockList'
import documentBlockMock from '@/mocks/DocumentBlock'
import createManyMocks from '@/mocks/createManyMocks'

storiesOf('DocumentEditor', module).add('status(undefined)', () => ({
  components: { TipeDocumentBlockList },
  data() {
    return {
      blocks: createManyMocks(
        () =>
          documentBlockMock({
            value: true,
            status: '',
            waiting: false,
            disabled: false
          }),
        3
      )
    }
  },
  template: '<tipe-document-block-list :blocks="blocks" />'
}))
