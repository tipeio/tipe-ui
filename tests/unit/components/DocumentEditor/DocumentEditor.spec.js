import { shallowMount } from '@vue/test-utils'
import DocumentEditor from '@/components/DocumentEditor'
import { mocks, fixtures } from '@tipe/tipe-test-utils'
import tipeTestUtils from '~/tests/TipeTestUtils'

describe('DocumentBlock', () => {
  describe('<template>', () => {
    tipeTestUtils.test.snapshot(DocumentEditor, {
      propsData: { document: fixtures.document() }
    })

    it('has correct data-tipe-iu attibute', () => {
      const propsData = { document: mocks.document() }
      const wrapper = shallowMount(DocumentEditor, { propsData })

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeDocumentEditor')
    })
  })
})
