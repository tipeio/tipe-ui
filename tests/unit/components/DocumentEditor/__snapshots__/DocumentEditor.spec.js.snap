// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentBlock <template> matches previous snapshot 1`] = `
<div data-tipe-ui="TipeDocumentEditor">
  <TipeDocumentNameInput-stub value="test-name"></TipeDocumentNameInput-stub>
  <TipeDocumentBlockList-stub blocks="[object Object],[object Object],[object Object]"></TipeDocumentBlockList-stub>
</div>
`;
